client/src/components/AddPost/AddPost.tsx

```diff
@@ -55,7 +55,7 @@ const AddPost = ({
   posts: Array<IPost>;
   setPost: Function;
 }) => {
-  const { user } = React.useContext(authContext);
+  const userContext = React.useContext(authContext);
```

Comment:
just destructure that and set open
```suggestion
  const { user, setOpen } = React.useContext(authContext);
```